digest/src/lib.rs

```diff
@@ -67,25 +67,25 @@ pub use digest_macro_derive::Digest;
 pub use hex;
 
 pub trait Digest {
-    type Hash: Hash + IntoDigestInput;
-    fn digest(self) -> Self::Hash;
+    type Hash: Hash + ToDigestInput;
```

Comment:
I actually didn't notice before, why are we requiring this trait bound here?

Reply:
Can you read the doc in the `lib.rs` and let me know if the question still holds?

Reply:
Makes sense now, thanks :)
I think I was mixing it up with the `Digest` trait somehow.